src/app/seller-post-ad-form/seller-post-ad-form.component.html

```diff
@@ -0,0 +1,129 @@
+            <div class="alert alert-danger" *ngIf="name.touched && !name.valid">
+                <div *ngIf="name.errors.required">Title is required</div>
+                <div *ngIf="name.errors.minlength">Title should be minimum 2 Characters.</div>
+            </div>
+        </div>
+
+        <div class="form-group">
+            <label>Category*</label>
+            <select ngModel id="category" class="form-control" name="category" required>
+            <option *ngFor="let category of categories">{{ category }}</option>
+            </select>
+        </div>
+
+        <div class="form-group">
+            <label for="price">Price*</label>
+            <div class="input-group">
+                <span class="input-group-text">₹</span>
+                <input ngModel id="price" name="price" #price="ngModel" class="form-control" type="number" placeholder="Product price.." required [min]="1">
+            </div>
+            <div class="alert alert-danger" *ngIf="price.touched && !price.valid">
+                <div *ngIf="price.errors.required"> Price is required.</div>
+                <div *ngIf="price.errors.min"> Price should be greater than .</div>
+            </div>
+        </div>
+
+        <div class="form-group">
+            <label for="discription ">Description*</label>
+            <textarea ngModel id="discription" minlength="4" name="discription" #discription="ngModel" class="form-control" type="text" placeholder="Product name.." required></textarea>
+            <div class="alert alert-danger" *ngIf="discription.touched && !discription.valid">
+                <div *ngIf="discription.errors.required">Description is required</div>
+                <div *ngIf="discription.errors.minlength">Description should be minimum 4 Characters.</div>
+            </div>
+        </div>
+
+        <div class="form-group">
+            <label for="dateOfPurchase">Date of purchase</label>
+            <input ngModel (change)="validateDate($event)" class="form-control" type="date" name="dateOfPurchase" required>
+            <div class="alert alert-danger" [style.display]="purchaseDate">
+                Date must not be greater than todays date.
+            </div>
+        </div>
+
+        <div class="form-group">
+            <label for="negotiable">Negotiable</label>
+            <div class="radio">
+                <label><input type="radio" ngModel name="isNegotiable" value="Yes">Yes</label>
+            </div>
+            <div class="radio">
+                <label><input type="radio"  ngModel name="isNegotiable" value="No" checked>No</label>
+            </div>
+        </div>
+
+
+        <div class="form-group">
+            <label style="margin-right: 20px;"> Address </label>
+            <input type="checkbox" (click)="validateAddress($event)" [value]='isAddressSelected'>
+            <div class="addresss" [style.display]="addressDisplayValue">
```

Comment:
Think of something to drive this using Boolean value which is better to understand.